File: Iteration-2/src/StudentId.java
import org.apache.log4j.Logger;

public class StudentId {
	private Logger log = Logger.getLogger(StudentId.class);
	private String cseCode;
	private int year;
	private int order;
	private String id;

	public StudentId(int year, int order) {
		this.cseCode = "1501";
		this.year = year;
		this.order = order;
		this.id = createStudentId();
		log.info("Student ID created.");
	}

	public String createStudentId() {
		return this.getCseCode() + Integer.toString(this.getYear()).substring(2) + "0"
				+ ((Integer.toString(this.getOrder()).length() == 2)
						? (Integer.toString(this.getOrder()))
						: ("0" + Integer.toString(this.getOrder())));
	}

	// GETTER & SETTER
	public String getCseCode() {
		return cseCode;
	}

	public int getYear() {
		return this.year;
	}

	public int getOrder() {
		return this.order;
	}

	@Override
	public String toString() {
		return id;
	}
}

File: Iteration-3/Code/StudentId.py
class StudentId:

    def __init__(self, year: int, order: int):
        self.__cseCode: str = "1501"
        self.__year: int = year
        self.__order: int = order
        self.__id: str = self.createStudentId()

    def createStudentId(self) -> str:
        order_length_dict = {1: "00", 2: "0", 3: ""}
        return self.__cseCode + str(self.__year)[-2:] + order_length_dict[len(str(self.__order))] + str(self.__order)

    def getCseCode(self) -> str:
        return self.__cseCode

    def getYear(self) -> int:
        return self.__year

    def getOrder(self) -> int:
        return self.__order

    def toString(self):
        return self.__id